### ISSUE TITLE
Dropdown component keyboard operation incorrect on Firefox

### ISSUE THREAD

**Issue by tomlyn** [carbon-design-system/carbon#7484]
On Firefox 78.6.0esr (64-bit),  the Dropdown component has an error when operated using the keyboard. 

1. Tab to the dropdown and opening the menu by pressing space bar
2. Use the down/up arrow keys to highlight the option you want
3. Press space bar to select the option
4. The menu closes and then immediately reopens. 

Note: You can press Esc to close the menu but the menu should not stay open after an option has been selected. 

The Dropdown component works as expected using the keyboard on Chrome.
 
This can be demonstrated on the Carbon Docs Dropdown component page
https://www.carbondesignsystem.com/components/dropdown/usage/#live-demo

![DropdownKeyboard](https://user-images.githubusercontent.com/15114439/102652859-f03aa200-4122-11eb-8b79-88f629038a4c.gif)

## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

> Is this issue related to a specific component?
Yes, the dropdown component

> What did you expect to happen? What happened instead? What would you like to
> see changed?

When I pressed the space bar I expected the option to be selected and for the menu to close -- like it does on Chrome.

> What browser are you working in?
Firefox 78.6.0esr (64-bit)

> What version of the Carbon Design System are you using?
10

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

Elyra Canvas. No pressing ship issues.

## Steps to reproduce the issue

See above - you can reproduce it in the Carbon docs.

This can be reproduced in the Carbon Docs Live Demo

**Comment by tomlyn** [comment]
Here's a better recording of the issue. After selecting various options with the space bar and the menu reopening I finally close the dropdown menu by pressing Esc.  

https://user-images.githubusercontent.com/15114439/102653182-78b94280-4123-11eb-95bc-c9741ce9df5e.mp4

**Comment by tw15egan** [comment]
@dakahn what is the expected keyboard interaction here? 

**Select** 
(Firefox)
To open: 
 <kbd>space</kbd>

To close:
<kbd>return</kbd> or <kbd>esc</kbd>.  <kbd>space</kbd> does nothing

(Chrome)
To open: 
 <kbd>space</kbd>

To close:
<kbd>return</kbd>,<kbd>esc</kbd>, and <kbd>space</kbd> 

**Dropdown**
(Firefox)
To open: 
 <kbd>space</kbd> or  <kbd>return</kbd>

To close:
<kbd>return</kbd> or <kbd>esc</kbd>,  <kbd>space</kbd> selects an item, closes and then reopens. 

(Chrome)
To open: 
 <kbd>space</kbd>

To close:
<kbd>return</kbd>,<kbd>esc</kbd>, and <kbd>space</kbd>

**Comment by dakahn** [comment]
list box item selection is done with the <kbd>enter</kbd> key as described in the WAI-Aria practices page for Collapsible Dropdown ListBox [seen here](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)

closing. i can reopen if there's something I missed.
